deployment/stacks/api.ts

```diff
@@ -0,0 +1,423 @@
+  aws_s3_notifications as s3notifications,
+  aws_wafv2 as wafv2,
+  Duration,
+  RemovalPolicy,
+  Tags,
+} from "aws-cdk-lib";
+import { Lambda } from "../constructs/lambda";
+import { WafConstruct } from "../constructs/waf";
+import { getSubnets } from "../utils/vpc";
+import { LambdaDynamoEventSource } from "../constructs/lambda-dynamo-event";
+import { DynamoDBTableIdentifiers } from "../constructs/dynamodb-table";
+import { isDefined } from "../utils/misc";
+import { isLocalStack } from "../local/util";
+
+// TODO: does this need to point to the tsconfig.json file in services/app-api?
+
+interface CreateApiComponentsProps {
+  scope: Construct;
+  stage: string;
+  project: string;
+  isDev: boolean;
+  vpc: ec2.IVpc;
+  kafkaAuthorizedSubnets: ec2.ISubnet[];
+  tables: DynamoDBTableIdentifiers[];
+  brokerString: string;
+  iamPermissionsBoundary: iam.IManagedPolicy;
+  iamPath: string;
+  wpFormBucket: s3.IBucket;
+  sarFormBucket: s3.IBucket;
+  templateBucket: s3.IBucket | undefined;
+}
+
+export function createApiComponents(props: CreateApiComponentsProps) {
+  const {
+    scope,
+    stage,
+    project,
+    isDev,
+    vpc,
+    kafkaAuthorizedSubnets,
+    tables,
+    brokerString,
+    iamPermissionsBoundary,
+    iamPath,
+    wpFormBucket,
+    sarFormBucket,
+    templateBucket,
+  } = props;
+
+  const service = "app-api";
+
+  const kafkaSecurityGroup = new ec2.SecurityGroup(
+    scope,
+    "KafkaSecurityGroup",
+    {
+      vpc,
+      description:
+        "Security Group for streaming functions. Egress all is set by default.",
+      allowAllOutbound: true,
+    }
+  );
+
+  const logGroup = new logs.LogGroup(scope, "ApiAccessLogs", {
+    removalPolicy: isDev ? RemovalPolicy.DESTROY : RemovalPolicy.RETAIN,
+  });
+
+  const api = new apigateway.RestApi(scope, "ApiGatewayRestApi", {
+    restApiName: `${stage}-app-api`,
+    deploy: true,
+    cloudWatchRole: false,
+    deployOptions: {
+      stageName: stage,
+      tracingEnabled: true,
+      loggingLevel: apigateway.MethodLoggingLevel.INFO,
+      dataTraceEnabled: true,
+      accessLogDestination: new apigateway.LogGroupLogDestination(logGroup),
+    },
+    defaultCorsPreflightOptions: {
+      allowOrigins: apigateway.Cors.ALL_ORIGINS,
+      allowMethods: apigateway.Cors.ALL_METHODS,
+    },
+  });
+
+  api.addGatewayResponse("Default4XXResponse", {
+    type: apigateway.ResponseType.DEFAULT_4XX,
+    responseHeaders: {
+      "Access-Control-Allow-Origin": "'*'",
+      "Access-Control-Allow-Headers": "'*'",
+    },
+  });
+
+  api.addGatewayResponse("Default5XXResponse", {
+    type: apigateway.ResponseType.DEFAULT_5XX,
+    responseHeaders: {
+      "Access-Control-Allow-Origin": "'*'",
+      "Access-Control-Allow-Headers": "'*'",
+    },
+  });
+
+  const environment = {
+    NODE_OPTIONS: "--enable-source-maps",
+    BOOTSTRAP_BROKER_STRING_TLS: brokerString,
+    stage,
+    TEMPLATE_BUCKET: templateBucket ? templateBucket.bucketName : "",
```

Comment:
I think could be
`templateBucket?.bucketName ?? ""`
Not sure that's actually better 🤷